superset-frontend/packages/superset-ui-core/src/query/extractTimegrain.ts

```diff
@@ -23,7 +23,7 @@ import { TimeGranularity } from '../time-format';
 
 export default function extractTimegrain(
   formData: QueryFormData,
-): TimeGranularity | undefined {
+): TimeGranularity | undefined | string {
```

Comment:
I am not excited about changing this to include strings, even though it seems to be pretty 50/50 in the codebase. Is it worth to go through and refactor the string portions @eschutho